digdag-tests/src/test/java/acceptance/PgIT.java

```diff
@@ -199,6 +199,73 @@ public void selectAndDownloadWithNullValues()
     }
 
     @Test
+    public void selectAndStoreResult()
+            throws Exception
+    {
+        copyResource("acceptance/pg/select_store_result.dig", root().resolve("pg.dig"));
+        copyResource("acceptance/pg/select_table.sql", root().resolve("select_table.sql"));
+
+        setupSourceTable();
+
+        CommandStatus status = TestUtils.main(
+                "run", "-o", root().toString(),
+                "--project", root().toString(),
+                "-p", "pg_database=" + tempDatabase,
+                "-p", "outfile=out",
+                "pg.dig");
+        assertThat(status.code(), is(0));
+
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(new File(root().toFile(), "out")))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+            assertThat(lines.toString(), is(stringContainsInOrder(
+                    Arrays.asList("foo", "bar", "baz")
+            )));
+        }
+    }
+
+    @Test
+    public void selectAndStoreResultWithExceedingMaxStoreResultRows()
+            throws Exception
+    {
+        copyResource("acceptance/pg/select_store_result.dig", root().resolve("pg.dig"));
+        copyResource("acceptance/pg/select_table.sql", root().resolve("select_table.sql"));
+
+        setupSourceTable();
+
+        CommandStatus status = TestUtils.main(
+                "run", "-o", root().toString(),
+                "--project", root().toString(),
+                "-p", "pg_database=" + tempDatabase,
+                "-p", "outfile=out",
+                "-X", "config.pg.max_store_result_rows=2",
+                "pg.dig");
+        assertThat(status.code(), is(1));
```

Comment:
It's better to check error message as well as exit code to confirm that tests don't wrongly success when pg operator failed to parse `config.pg.max_store_result_rows` option.

Reply:
👍 
Depending on an error message is a bit naive, but it's better than never. Maybe we'd better add error codes more reliable in the future.